ext/re2/re2.cc

```diff
@@ -1362,6 +1375,152 @@ static VALUE re2_QuoteMeta(VALUE self, VALUE unquoted) {
+}
+
+/*
+ * Returns a new {RE2::Set} object.
+ *
+ * @return [RE2::Set]
+ *
+ * @overload initialize
+ *   Returns a new {RE2::Set} object for unanchored patterns with the default
+ *   options.
+ *
+ * @overload initialize(anchor)
+ *   Returns a new {RE2::Set} object for the specified anchor with the default
+ *   options.
+ *
+ *   @raise [ArgumentError] if anchor is not one of the accepted choices
+ *   @raise [NoMemoryError] if memory could not be allocated for the compiled pattern
+ *
+ * @overload initialize(anchor, options)
+ *   Returns a new {RE2::Set} object with the specified options.
+ *
+ *   @param [Symbol] anchor One of :unanchored, :anchor_start, :anchor_both
+ *   @param [Hash] options the options with which to compile the pattern
+ *   @option options [Boolean] :utf8 (true) text and pattern are UTF-8; otherwise Latin-1
+ *   @option options [Boolean] :posix_syntax (false) restrict regexps to POSIX egrep syntax
+ *   @option options [Boolean] :longest_match (false) search for longest match, not first match
+ *   @option options [Boolean] :log_errors (true) log syntax and execution errors to ERROR
+ *   @option options [Fixnum] :max_mem approx. max memory footprint of RE2
+ *   @option options [Boolean] :literal (false) interpret string as literal, not regexp
+ *   @option options [Boolean] :never_nl (false) never match \n, even if it is in regexp
+ *   @option options [Boolean] :case_sensitive (true) match is case-sensitive (regexp can override with (?i) unless in posix_syntax mode)
+ *   @option options [Boolean] :perl_classes (false) allow Perl's \d \s \w \D \S \W when in posix_syntax mode
+ *   @option options [Boolean] :word_boundary (false) allow \b \B (word boundary and not) when in posix_syntax mode
+ *   @option options [Boolean] :one_line (false) ^ and $ only match beginning and end of text when in posix_syntax mode
+ *   @return [RE2::Set] an RE2::Set with the specified anchor and options
+ *   @raise [ArgumentError] if anchor is not one of the accepted choices
+ *   @raise [NoMemoryError] if memory could not be allocated for the compiled pattern
+ */
+static VALUE re2_set_initialize(int argc, VALUE *argv, VALUE self) {
+  VALUE anchor, options;
+  re2_set *s;
+  RE2::Anchor re2_anchor;
+  RE2::Options re2_options;
+
+  rb_scan_args(argc, argv, "02", &anchor, &options);
+  Data_Get_Struct(self, re2_set, s);
+
+  if (RTEST(options)) {
+    re2_options = parse_re2_options(options);
+  } else {
+    re2_options = RE2::DefaultOptions;
+  }
+  if (NIL_P(anchor)) {
+    re2_anchor = RE2::UNANCHORED;
+  } else {
+    Check_Type(anchor, T_SYMBOL);
+    ID id_anchor = SYM2ID(anchor);
+    if (id_anchor == id_unanchored) re2_anchor = RE2::UNANCHORED;
+    else if (id_anchor == id_anchor_start) re2_anchor = RE2::ANCHOR_START;
+    else if (id_anchor == id_anchor_both) re2_anchor = RE2::ANCHOR_BOTH;
+    else {
+      rb_raise(rb_eArgError, "anchor should be one of: :unanchored, :anchor_start, :anchor_both");
+    }
```

Comment:
> use a `switch`?

`switch` requires `case` predicates to be compile-time constants. I tried but found no way to make the ids `const`, since `rb_intern()` cannot be a `constexpr`-fn.

> could we prefer conditionals with braces

This shall be done.